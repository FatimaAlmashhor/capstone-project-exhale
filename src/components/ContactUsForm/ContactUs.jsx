import React, { useState } from 'react';
import emailjs from 'emailjs-com';
import { useTranslation } from 'react-i18next';

export default function ContactUs() {
  const { t } = useTranslation();
  const [success, setSuccess] = useState('');
  const [errMessage, setErrMessage] = useState('');

  function sendEmail(e) {
    e.preventDefault();

    emailjs
      .sendForm(
        'gmail_Exhale',
        'template_g1p37uk',
        e.target,
        'user_ux4cNdoe5I82QS9ztaxr7'
      )
      .then(
        () => {
          setErrMessage('');
          setSuccess('You message have been successfully sent.');
        },
        (error) => {
          setSuccess('');
          setErrMessage(error.message);
        }
      );
  }

  return (
    <section className="text-gray-700 body-font ">
      <div className="container px-5 py-24 mx-auto">
        <div className="flex flex-col w-full mb-12 text-center">
          <h1 className="mb-4 text-2xl font-medium text-blue-800 sm:text-3xl title-font">
            {t('GetInTouch')}
          </h1>
          <p className="mx-auto text-base leading-relaxed lg:w-2/3">
            {t('contactus')}
          </p>
        </div>
        <form className="contact-form" onSubmit={sendEmail}>
          <div className="mx-auto lg:w-1/2 md:w-2/3">
            <div className="flex flex-wrap -m-2">
              <div className="w-1/2 p-2">
                <input type="hidden" name="contact_number" />
                <input
                  type="text"
                  name="user_name"
                  placeholder={t('Name')}
                  className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
                />
              </div>
              <div className="w-1/2 p-2">
                <input
                  type="email"
                  name="user_email"
                  placeholder={t('Email')}
                  className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
                />
              </div>
              <div className="w-full p-2">
                <textarea
                  name="message"
                  placeholder={t('Message')}
                  className="w-full h-32 px-3 py-1 text-base leading-6 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none resize-none focus:border-indigo-500"
                />
              </div>
              <div className="w-full p-2">
                <input
                  type="submit"
                  value={t('Send')}
                  className="inline-block w-full px-4 py-3 mt-5 text-white bg-blue-800 rounded shadow-lg hover:bg-blue-600 focus:bg-blue-700"
                />
              </div>
            </div>
          </div>
        </form>
        <div className="text-lg text-red-500">
          {success}
          {errMessage}
        </div>
      </div>
    </section>
  );
}
