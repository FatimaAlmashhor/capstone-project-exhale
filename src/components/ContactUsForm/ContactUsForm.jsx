import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import emailjs from 'emailjs-com';
import Lottie from 'react-lottie';
import received from '../../Lottie/message-received.json';
import Modal, { Body } from '../common/Modal';

export default function ContactUsForm() {
  const { t } = useTranslation();

  const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: received,
    rendererSettings: {
      preserveAspectRatio: 'xMidYMid slice',
    },
  };

  const [show, setShow] = useState(false);
  const [errMsg, setErrMsg] = useState('');

  const handleTarget = (e) => {
    e.preventDefault();
    emailjs
      .sendForm(
        'Exhale_Project',
        'template_yu5s23r',
        e.target,
        'user_Uu00ojF2R1FaqDmvnK8bO'
      )
      .then(
        () => {
          setShow(true);
          setErrMsg('');
        },
        (error) => {
          setErrMsg(error.text);
        }
      );
    e.target.reset();
  };

  return (
    <section className="text-gray-700 body-font">
      <div className="container px-5 py-24 mx-auto">
        <div className="flex flex-wrap justify-center py-8 text-center ">
          <div className="w-full px-12 md:w-6/12 md:px-4">
            <h2 className="text-4xl text-blue-700 ">{t('GetInTouch')}</h2>
            <p className="mt-4 mb-4 text-lg leading-relaxed text-gray-600">
              {t('contactus')}
            </p>
          </div>
        </div>
        <form onSubmit={handleTarget}>
          <div className="mx-auto lg:w-1/2 md:w-2/3">
            <div className="flex flex-wrap -m-2">
              <div className="w-1/2 p-2">
                <input
                  type="text"
                  id="name"
                  name="name"
                  placeholder={t('Name')}
                  className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
                  required
                />
              </div>
              <div className="w-1/2 p-2">
                <input
                  type="email"
                  id="email"
                  name="email"
                  placeholder={t('Email')}
                  className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
                  required
                />
              </div>
              <div className="w-full p-2">
                <textarea
                  id="message"
                  name="message"
                  placeholder={t('Message')}
                  className="w-full h-32 px-3 py-1 text-base leading-6 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none resize-none focus:border-indigo-500"
                  required
                />
              </div>
              <div className="w-full p-2">
                <button
                  type="submit"
                  className="inline-block w-full px-4 py-3 mt-5 text-white bg-blue-800 rounded shadow-lg hover:bg-blue-600 focus:bg-blue-700"
                >
                  {t('Send')}
                </button>
              </div>
            </div>
          </div>
          <div className="text-lg text-center text-red-500">{errMsg}</div>
        </form>
      </div>
      {show ? (
        <>
          <Modal
            show={show}
            onClick={() => {
              setShow(false);
            }}
          >
            <div>
              <Body>
                <div>
                  <div>
                    <h1 className="text-xl text-center text-teal-500">
                      {t('Thank you for contacting us')}
                    </h1>
                    <div className="relative w-64 h-64 mx-auto overflow-hidden">
                      <Lottie
                        isClickToPauseDisabled
                        options={defaultOptions}
                        height="auto"
                        width="100%"
                      />
                    </div>
                    <div className="w-full mt-5 ">
                      <p className="text-lg text-center text-blue-800">
                        {t('Your message have been received successfully.')}
                      </p>
                    </div>
                  </div>
                </div>
              </Body>
            </div>
          </Modal>
        </>
      ) : null}
    </section>
  );
}
