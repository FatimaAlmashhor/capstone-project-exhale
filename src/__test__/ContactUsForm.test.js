import 'jest-canvas-mock';
import React from 'react';
import renderer from 'react-test-renderer';
import ContactUsForm from '../components/ContactUsForm';

it('renders correctly', () => {
  const tree = renderer.create(<ContactUsForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
