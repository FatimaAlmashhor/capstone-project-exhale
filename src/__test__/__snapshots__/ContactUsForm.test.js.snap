// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="text-gray-700 body-font"
>
  <div
    className="container px-5 py-24 mx-auto"
  >
    <div
      className="flex flex-wrap justify-center py-8 text-center "
    >
      <div
        className="w-full px-12 md:w-6/12 md:px-4"
      >
        <h2
          className="text-4xl text-blue-700 "
        >
          GetInTouch
        </h2>
        <p
          className="mt-4 mb-4 text-lg leading-relaxed text-gray-600"
        >
          contactus
        </p>
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="mx-auto lg:w-1/2 md:w-2/3"
      >
        <div
          className="flex flex-wrap -m-2"
        >
          <div
            className="w-1/2 p-2"
          >
            <input
              className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
              id="name"
              name="name"
              placeholder="Name"
              required={true}
              type="text"
            />
          </div>
          <div
            className="w-1/2 p-2"
          >
            <input
              className="w-full px-3 py-1 text-base leading-8 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none focus:border-indigo-500"
              id="email"
              name="email"
              placeholder="Email"
              required={true}
              type="email"
            />
          </div>
          <div
            className="w-full p-2"
          >
            <textarea
              className="w-full h-32 px-3 py-1 text-base leading-6 text-gray-700 transition-colors duration-200 ease-in-out bg-gray-100 border border-gray-300 rounded outline-none resize-none focus:border-indigo-500"
              id="message"
              name="message"
              placeholder="Message"
              required={true}
            />
          </div>
          <div
            className="w-full p-2"
          >
            <button
              className="inline-block w-full px-4 py-3 mt-5 text-white bg-blue-800 rounded shadow-lg hover:bg-blue-600 focus:bg-blue-700"
              type="submit"
            >
              Send
            </button>
          </div>
        </div>
      </div>
      <div
        className="text-lg text-center text-red-500"
      >
        
      </div>
    </form>
  </div>
</section>
`;
